### ISSUE TITLE
extra toolbar buttons on wysiwyg title fields e.g. text alignment?

### ISSUE THREAD

**Issue by ronm123** [thexerteproject/xerteonlinetoolkits#601]
Now that it's possible to drag and drop images onto title fields or have FA icons etc it would be useful to allow control of text alignment too. Another example is the new caption option on the graphics and sound page where rather than have an additional optional property to align the caption centre or left having the toolbar options for that would be far simpler and more universal. Appreciate that this might make it difficult to keep that toolbar to a single line but if that's not a problem then I think would be a useful  enhancement.

**Comment by torinfo** [comment]
It will remain a compromise between keeping this a one line menu and functionlaity..... But it is not hard to do.

**Comment by FayCross** [comment]
I'd like to add the text alignment options (someone here has asked for it). I can get the buttons to appear on the toolbar and it seems to work ok...

![wysiwyg1](https://user-images.githubusercontent.com/8926470/26875138-699c74d8-4b78-11e7-94d1-86f2d015eaf5.png)

...but then if you then change page in the editor & go back it has added extra p tags too...

![wysiwyg2](https://user-images.githubusercontent.com/8926470/26875148-7460c02c-4b78-11e7-81f0-1d0a8687753e.png)

Any idea why this would happen @torinfo? All I have done is add JustifyLeft, JustifyCenter & JustifyRight to the list of what's included in the menu in toolbox.js

**Comment by torinfo** [comment]
Hmmm...

The problem is that we manipulate the results of the editor TextInputs in a way that is not working in this case.

The editor aways wraps the text in <p> tags, and we remove those. Because of the style attribute that obviously doesn't work.

What would be better (I think) is to add a class to the <p> tags that makes sure that for that paragraph the margins, padding etc are zero instead of removing the tag completely.

So instead of removing and adding the <p> tag we add/remove a class to the <p> tag.

**Comment by FayCross** [comment]
Sorry I should have commented again as earlier as I found where the p tags were being added in the editor & have changed this slightly so that if the p tag is already there then it doesn't wrap it in p tags again. It seems to work ok but if there's a better way of dealing with this with classes then feel free to change it. Thanks

**Comment by torinfo** [comment]
Regrettably this will solve the issue in the TextInputs itself, but now, you have a <p> around the text which can result in strange extra margins/padding in some of the pages.

They <p> will not be removed, because the code is explicitly looking for '<p>' and not '<p.*>'

**Comment by FayCross** [comment]
@torinfo maybe we can add the alignment buttons to the small wysiwyg toolbar now that you have fixed the issue with extra lines being added  (https://github.com/thexerteproject/xerteonlinetoolkits/issues/1261#event-10864905350)

**Comment by torinfo** [comment]
Yes, I'll have a look at this. It might result in small changes in how the content looks, because now in some cases the code will be wrapped in <p> and in some case not... But it really depends on the themes.
